sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -1118,35 +1207,38 @@ public new AppendBlobClient WithEncryptionScope(string encryptionScope)
         /// </remarks>
         internal async Task<Response<BlobAppendInfo>> AppendBlockInternal(
             Stream content,
-            byte[] transactionalContentHash,
-            AppendBlobRequestConditions conditions,
-            IProgress<long> progressHandler,
+            AppendBlobAppendBlockOptions options,
             bool async,
             CancellationToken cancellationToken)
         {
             using (ClientConfiguration.Pipeline.BeginLoggingScope(nameof(AppendBlobClient)))
             {
+                options ??= new AppendBlobAppendBlockOptions();
+
                 ClientConfiguration.Pipeline.LogMethodEnter(
                     nameof(AppendBlobClient),
                     message:
                     $"{nameof(Uri)}: {Uri}\n" +
-                    $"{nameof(conditions)}: {conditions}");
+                    $"{nameof(options.Conditions)}: {options.Conditions}");
 
                 DiagnosticScope scope = ClientConfiguration.ClientDiagnostics.CreateScope($"{nameof(AppendBlobClient)}.{nameof(AppendBlock)}");
 
                 // All AppendBlobRequestConditions are valid.
-                conditions.ValidateConditionsNotPresent(
+                options.Conditions.ValidateConditionsNotPresent(
```

Comment:
`ValidateConditionsNotPresent()` works on null conditions, IIRC.  Is this correct?

Reply:
Correct. The extension method does a null check on the `this` parameter.